packages/angular/src/jsonforms.service.ts

```diff
@@ -82,25 +132,62 @@ export class JsonFormsAngularService {
         return coreAction;
     }
 
+    /**
+     * @deprecated use {@link JsonFormsAngularService.setUiSchemas}
+     */
     updateUiSchema<T extends UISchemaActions>(uischemaAction: T): T {
         const uischemaState = uischemaRegistryReducer(this._state.uischemas, uischemaAction);
         this._state.uischemas = uischemaState;
         this.updateSubject();
         return uischemaAction;
     }
 
+    setUiSchemas(uischemas: { tester: UISchemaTester; uischema: UISchemaElement; }[]): void {
+        this._state.uischemas = uischemas;
+        this.updateSubject();
+    }
+
     updateConfig<T extends SetConfigAction>(setConfigAction: T): T {
         const configState = configReducer(this._state.config, setConfigAction);
         this._state.config = configState;
         this.updateSubject();
         return setConfigAction;
     }
 
+    setUiSchema(uischema: UISchemaElement): void {
+        const newUiSchema = uischema ?? generateDefaultUISchema(this._state.core.schema);
+        const coreState = coreReducer(this._state.core, Actions.updateCore(this._state.core.data, this._state.core.schema, newUiSchema));
+        this._state.core = coreState;
+        this.updateSubject();
+    }
+
+    setSchema(schema: JsonSchema): void {
+        const coreState = coreReducer(this._state.core, Actions.updateCore(this._state.core.data, schema, this._state.core.uischema));
```

Comment:
In `setUiSchema` we generate the ui schema when the user hands over `undefined`. We should do the same here.